sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/EventProcessor.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Messaging.EventHubs.Core;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    /// <summary>
+    ///   Constantly receives <see cref="EventData" /> from every partition in the context of a given consumer group.
+    ///   The received data is sent to an <see cref="IPartitionProcessor" /> to be processed.
+    /// </summary>
+    ///
+    public class EventProcessor
+    {
+        /// <summary>
+        ///   A unique name used to identify this event processor.
+        /// </summary>
+        ///
+        public string Name { get; }
```

Comment:
Did we settle on `Name` or `InstanceId` or `Id` or `Identifier`?
Check with the other languages

Reply:
As far as I know, we haven't discussed about this. It seems Python and Java are using `instance id`, whereas Typescript calls it `id` only. I'll have it changed to `InstanceId` for now.